src/service.c

```diff
@@ -1709,7 +1709,20 @@ static void svc_set_state(svc_t *svc, svc_state_t new)
 	if (svc_is_runtask(svc)) {
 		char cond[MAX_COND_LEN];
 
-		snprintf(cond, sizeof(cond), "%s/%s/done", svc->type, svc->name);
+		switch (svc->type) {
+		case SVC_TYPE_TASK:
+			snprintf(cond, sizeof(cond), "task/%s/done", svc->name);
+			break;
+		case SVC_TYPE_RUN:
+			snprintf(cond, sizeof(cond), "run/%s/done", svc->name);
+			break;
+		case SVC_TYPE_SYSV:
+			snprintf(cond, sizeof(cond), "sysv/%s/done", svc->name);
+			break;
```

Comment:
Good catch, I didn't spot this in my review, sorry about that!

In general I try to avoid spreading out hard coded constants.  Could you please change the above to:
```C
  case SVC_TYPE_TASK:
  case SVC_TYPE_RUN:
  case SVC_TYPE_SYSV:
     snprintf(cond, sizeof(cond), "%s/%s/done", svc_typestr(svc), svc->name);
     break;
```

Reply:
@troglobit 

Arha, good suggestion, then maybe the switch/case could be dropped as well, just use "svc_typestr(svc)", will send a V2.

Reply:
Well maybe, it served as a neat filter -- only task/run/sysv can create the done condition.  But either way is fine with me :)